docs/configuration/extract-codacy-logs.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+
+#### Helper functions ####
+
+usage()
+{
+  echo "Usage: $0 -n NAMESPACE "
+  exit 2
+}
+
+set_variable()
+{
+  local varname=$1
+  shift
+  if [ -z "${!varname}" ]; then
+    eval "$varname=\"$@\""
+  else
+    echo "Error: $varname already set"
+    usage
+  fi
+}
+
+
+#### Log extraction script ####
+
+while getopts "n:" option; do
+	case ${option} in
+		n )
+			set_variable NAMESPACE ${OPTARG}
+			;;
+		\? )
+			usage
+			;;
+	esac
+done
+
+[ -z "$NAMESPACE" ] && usage
+
+CURRENT_DATE_TIME=$(date "+%Y%m%d-%H%M%S")
+LOGS_DIR=/tmp/codacy_logs_$CURRENT_DATE_TIME
+
+echo "Starting log files extraction"
+
+# Check if kubectl is available
+echo "Checking if kubectl is installed..."
+kubectl --version &>/dev/null
+if [ $? -ne 0 ]; then
+	echo "kubectl not installed"
+	echo "Please install kubectl version specified in Codacy's documentation (see here - https://codacy.github.io/chart/#preparing-to-install-codacy) or the version used when installing your cluster"
+	echo "To install kubectl see https://kubernetes.io/docs/tasks/tools/install-kubectl/"
+	echo "Log files extraction failed, exiting..."
+	exit $?
+fi
+
+# Check current cluster context
+echo "Checking access to kubernetes cluster..."
+$KUBE_CTX=$(kubectl config current-context)
+if [ $? -ne 0 ]; then
+	echo "No kubernetes cluster context configured"
+	echo "Log files extraction failed, exiting..."
+	exit $?
+fi
+
+echo "Is '$KUBE_CTX' the correct kubernetes cluster for log extraction? (yes/no)"
+read -p ANSWER
+if [ $ANSWER != "yes" ]; then
+	echo "Please configure correctly your current kubernetes cluster"
+	echo "Log files extraction failed, exiting..."
+	exit 2
+fi
+
+# Create temporary directory for copying logs
+mkdir $LOGS_DIR
+if [ $? -ne 0 ]; then
+	echo "Failed to create temporary directory $LOGS_DIR , for log files extraction"
```

Comment:
If this happens we should probably continue in the local folder.

Reply:
hmm, I would say if this happens the machine is fucked up.
What would be another reason for a failure here?

Reply:
Maybe some permissions issue? Not sure when you don't have permissions to write to `/tmp` so let's not try to be too smart about this. I'll mark it as resolved.